### ISSUE TITLE
MySQL ON DELETE CASCADE is not supported on FOREIGN KEYS

### ISSUE THREAD

**Issue by sy6sy2** [sqlfluff/sqlfluff#2449]
SQLFluff report `L014 | Unquoted identifiers must be consistently lower case.` on the last two lines of this:

```
CREATE TABLE source_tag_assoc (
    source_id INT UNSIGNED NOT NULL,
    tag_id INT UNSIGNED NOT NULL,
    PRIMARY KEY (source_id, tag_id),
    FOREIGN KEY (source_id) REFERENCES source (id) ON DELETE CASCADE,
    FOREIGN KEY (tag_id) REFERENCES source_tag (id) ON DELETE CASCADE
);
```

but I don't see any problem about cases here.

## Expected Behaviour

No warning on this.

## Observed Behaviour

L014 warning.

## Steps to Reproduce

Run `sqlfluff lint --dialect mysql` on the given example.

## Dialect

Mysql

## Version

```
sqlfluff, version 0.9.2
Python 3.9.9
```

## Configuration
```
Include your SQLFluff configuration here
```

macOS 12.1

**Comment by tunetheweb** [comment]
The is a case of SQLFluff giving out the incorrect error message because it cannot understand the query.

The problem is `FOREIGN` is being flagged as an identifier rather than a keyword:

```
% sqlfluff parse --dialect mysql test.sql
...
[L:  4, P:  5]      |                table_constraint_segment:
[L:  4, P:  5]      |                    keyword:                                  'PRIMARY'
[L:  4, P: 12]      |                    whitespace:                               ' '
[L:  4, P: 13]      |                    keyword:                                  'KEY'
...
[L:  5, P:  5]      |                column_definition:
[L:  5, P:  5]      |                    identifier:                               'FOREIGN'
[L:  5, P: 12]      |                    whitespace:                               ' '
[L:  5, P: 13]      |                    data_type:
[L:  5, P: 13]      |                        data_type_identifier:                 'KEY'
...
[L:  6, P:  5]      |                column_definition:
[L:  6, P:  5]      |                    identifier:                               'FOREIGN'
[L:  6, P: 12]      |                    whitespace:                               ' '
[L:  6, P: 13]      |                    data_type:
[L:  6, P: 13]      |                        data_type_identifier:                 'KEY'
...
```

Therefore it says some of your identifiers are UPPERCASE and some lowercase, and hence rule L014 flags.

And the reason that it thinks `FOREIGN` is an identifier, rather than a keyword, is because it cannot match to this to the `TableConstraintSegment` because `ON DELETE CASCADE` has not been implemented in SQLFluff ror MySQL. If you remove that, then you can see it no longer throws the rule.

I've updated the issue title.

**Comment by sy6sy2** [comment]
Thank you very munch for your answer and the explanation!
Is the `ON DELETE CASCADE` support on the road?

**Comment by tunetheweb** [comment]
Added this in #2461
